llvm/lib/Transforms/Vectorize/VPlanValue.h

```diff
@@ -110,13 +110,18 @@ class LLVM_ABI_FOR_TEST VPValue {
   void dump() const;
 #endif
 
-  unsigned getNumUsers() const { return Users.size(); }
-  void addUser(VPUser &User) { Users.push_back(&User); }
+  /// Returns the number of uses of this VPValue. The same user could appear
+  /// more than once in the list due to this value being used for multiple
+  /// operands in a recipe.
+  unsigned getNumUses() const { return Users.size(); }
 
-  /// Remove a single \p User from the list of users.
-  void removeUser(VPUser &User) {
+  /// Adds a use of this value to the list.
+  void addUse(VPUser &User) { Users.push_back(&User); }
```

Comment:
Hm, this is clearer for some uses, but is inconsistent in other ways, as this doesn't really add a use, it adds a user object of which there are may be multiple and there's only accessors for iterating over users not uses at the moment

Reply:
IR/Value.h only has a single UseList and it creates iterators for both uses and users from the same UseList. Also, Value.h only has an `addUse` function - `addUser` does not exist. I know in other parts of the vectoriser we've tried hard to keep consistent naming schemes and terminology with the IR equivalents, which is what I'm trying to do here. Is the problem you're worried about here that `addUse` takes a `VPUser`?

Reply:
But what about `getNumUsers`? Are you saying that in VPlan a `User` is actually more like a `Use`? There is ambiguity when `getNumUsers` returns a value of 2 or more, since the caller has to figure out if this the same user using the value multiple times (for different operands) or different unique users. Would it be useful at least to have both `getNumUses` and `getNumUsers` where the latter actually does the work to figure out how many unique users?

Reply:
I've reverted addUser back to addUser.